### PR TITLE
plotly_click evt now carries pressed button information

### PR DESCRIPTION
**Pull request by aburato**
Add information about the pressed button in plotly_click event.

### CODE REVIEW

File: src/plots/cartesian/graph_interact.js [hunk 1]
```diff
@@ -2099,8 +2099,7 @@ fx.dragElement = function(options) {
         if(options.doneFn) options.doneFn(gd._dragged, numClicks);
 
         if(!gd._dragged) {
-            var e2 = document.createEvent('MouseEvents');
-            e2.initEvent('click', true, true);
+            var e2 = new MouseEvent('click', { 'button': e.button });
```
Comment (mdtusz) [inline src/plots/cartesian/graph_interact.js]: I don't believe event constructors like this (`new MouseEvent(..)`) [aren't fully supported in all browsers yet](https://msdn.microsoft.com/library/dn905219(v=vs.85).aspx), so we should stay with the "old" style.

Comment (etpinard) [inline src/plots/cartesian/graph_interact.js]: The *new style* is supported in IE9, according to https://developer.mozilla.org/en/docs/Web/API/MouseEvent

if that's the case, then :+1: for the new style.

Comment (mdtusz) [inline src/plots/cartesian/graph_interact.js]: Ah perfect! Seems to be working in Safari despite the `?` too so I guess Edge is the only unknown. :+1:

Comment (aburato) [inline src/plots/cartesian/graph_interact.js]: Sorry guys, I forgot you have to support crappy IE9, next time I'll make sure to comply with that as well. Anyways, if it works... :)

Comment (mdtusz) [inline src/plots/cartesian/graph_interact.js]: No apologies! We're always happy to have pull requests come in!

**Comment by mdtusz** [comment]
The code looks fine here, but we're a bit curious what the use-case for having the button value in events is? `plotly_click` is fired only on left-clicks (as far as I know) so the button value isn't too useful here.

**Comment by aburato** [comment]
@mdtusz nope, they get fired on right click as well, please look at the full commit. :) The use case is handling custom contextual actions (e.g: context menu for quick options).

**Comment by etpinard** [comment]
@aburato My apologies, but I can't get `plotly_click` to fire on right-click. 

Do you have a reproducible example? Or better yet, could you add a test case in [`click_test.js`](https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/click_test.js) ?

**Comment by aburato** [comment]
I see. Well, I'm quite in a hurry as I have to leave the office for the weekend, so here's a VERY dirty solution: an html sample inside the dist file. On latest Chrome, windows, right clicking on the chart works and yields "2" as the button.

Sorry for the unhortodox sample, I'll clean up things ASAP. In the mantime, feel free to ignore this if you want :)

**Comment by aburato** [comment]
Hi guys, sorry for the delay but I was on holidays :)

I cleaned up the dist folder and bundled the sample in this ZIP file:

https://drive.google.com/file/d/0B_fJz_aNFtqDZDRyMnBSYTh6Yms/view?usp=sharing

could you please give it a spin and see if it works for you?

Ciao!

**Comment by mdtusz** [comment]
Ahh I see now. To have the right click registered properly however (on chrome at least) you need to also have `document.oncontextmenu = function(){ return false; };` to disable the right click menu.

I see no issue with this, but ultimately it will be up to @etpinard and @alexcjohnson. 

In any case, we'll ask you to fix the conflicts and ensure that all tests are passing (ideally, write a few tests for the click event as well!) before we will merge this.

**Comment by aburato** [comment]
I don't know why the CircleCI tests are failing on the image compares... the button info added to the click event shouldn't have any affect on those tests... FWIW, I tried running them against the pristine point from which I branched, and those tests fail as well, so maybe I chose the wrong place to branch from :)

**Comment by etpinard** [comment]
@aburato 

In order for this PR to be approved, you'll need to add the same right-button functionality to our `geo` and `gl3d` plot types. We've made big efforts in the last few months to bring click interactions on par for all plot types and I wouldn't widen that gap again.

Moreover,

1. Don't commit `dist/` bundles to your PRs. We bundle our code only we bump the package [version](https://github.com/plotly/plotly.js/blob/bbb8205903cdd34ae019d5cd6a2c197e9a2550e7/package.json#L41).
1. I suspect that CircleCI is caching some of our dependencies and not bundling the most up-to-date package before running our image tests. Merging the upstream `master` branch should solve the problem.

**Comment by aburato** [comment]
@etpinard thanks! I do agree that the click functionality should be kept on the same level across all plot types. I'm not familiar with those plots, but I'll see what I can do.
Re. the dist commits, sorry, my mistake., won't happen again (and of course doesn't make any sense).

**Comment by etpinard** [comment]
Closing due to lack of activity.

@aburato feel free to open an [issue](https://github.com/plotly/plotly.js/issues/new) on the topic.

**Pull request closed without merging**